modules/nf-core/bowtie2/align/main.nf

```diff
@@ -38,6 +38,7 @@ process BOWTIE2_ALIGN {
     }
 
     def samtools_command = sort_bam ? 'sort' : 'view'
+    def extension = samtools_command == 'sort' ? 'bam' : 'sam'
```

Comment:
Can you get the extension from the `args2` instead? `samtools view` and `samtools sort` both have the `--output-fmt SAM|BAM|CRAM` option from which you can easily define what the output format should be

Reply:
Ah, I completely missed this option. My bad, of course, this makes more sense. Closing the PR then.